File: 2026-2/sesion-de-clase-4/dotnet/Lab01-2026-1/Assessment.cs
namespace Lab01_2026_1;

public class Assessment
{
    private readonly List<AssessmentItem> items;

    public int Duracion { get; }
    public DateTime FechaHora { get; }
    public IList<AssessmentItem> Items => items.AsReadOnly();
    public int Puntaje { get; set; }

    public Assessment(
            int duracion,
            DateTime fechaHora,
            List<AssessmentItem> items)
    {
        Duracion = duracion;
        FechaHora = fechaHora;
        this.items = [.. items];
    }
}

File: 2026-2/sesion-de-clase-04/java/Solución/Lab01-2026-1/pe/edu/pucp/examenes/Assessment.java
package pe.edu.pucp.examenes;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Assessment {

    private int duracion;
    private LocalDateTime fechaHora;
    private List<AssessmentItem> items;
    private int puntaje;

    public Assessment(int duracion,
            LocalDateTime fechaHora,
            List<AssessmentItem> items) {

        this.duracion = duracion;
        this.fechaHora = fechaHora;
        this.items = new ArrayList<>(items);
    }

    public int getDuracion() {
        return duracion;
    }

    public LocalDateTime getFechaHora() {
        return fechaHora;
    }

    public List<AssessmentItem> getItems() {
        return Collections.unmodifiableList(items);
    }

    public void setPuntaje(int puntaje) {
        this.puntaje = puntaje;
    }
}
